### PR TITLE
Logging style messages and critical telemetry events.

### LINKED ISSUE #632: Implement logging style messages in the loading screen.
### Dev Home version

_No response_

### Windows build number

_No response_

### Other software

_No response_

### Steps to reproduce the bug

Action center text is tertiary color.
Tasks should scroll and add one at a time when the are started
Primary when they are running.
Add tasks as they start to the bottom of the list
All messages on the task side should be secondary color.

### Expected result

_No response_

### Actual result

_No response_

### Included System Information

_No response_

### Included Extensions Information

_No response_

### LINKED ISSUE #1308: Update RepoTool_AddRepos_Event to critical event
### Suggested new feature or improvement

We are trying to understand and analyze the drop off point of the DevDrive creation from DevHome App.
As the DevDerive creation related events are critical, we would like to make the previous step event critical as well. Then the  creation process events' data could match each other.

We will create the Funnel for DevDrive creation based on the events in the creation process
1. RepoTool_AddRepos_Event
2. CreateDevDrive_CreatingDevDrive_Event
3. CreateDevDriveTriggered

### Scenario

Update RepoTool_AddRepos_Event from measure to critical event

### Additional details

_No response_

### PR DESCRIPTION
**Pull request by dhoehna**
## Summary of the pull request
All secondary style messages are inserted into the list instead of being changed in-place.

Changed all events to Critical.  I did get permission for this.

## Detailed description of the pull request / Additional comments
New ViewModel to handle loading screen messages.

Added code to facilitate inserting secondary messages at the index before any primary messages.

## PR checklist
- [x] Closes #1308 
- [x] Closes #632
- [x] Tests added/passed Yes and yes.
- [ ] Documentation updated N/A

Movie!
LoggingStyleMessagesButBetter

### CODE REVIEW

File: telemetry/DevHome.Telemetry/Telemetry.cs [hunk 1]
```diff
@@ -184,9 +184,9 @@ public void LogTimeTaken(string eventName, uint timeTakenMilliseconds, Guid? rel
     /// <param name="eventName">The name of the event to log</param>
```
Comment (krschau) [inline telemetry/DevHome.Telemetry/Telemetry.cs:184]: It won't let me comment on line 182, but if you removed "informal" above, do you want to remove it here too?

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/LoadingMessageViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    public string MessageToShow { get; }
+
+    /// <summary>
+    /// If the progress ring should be shown.  Only show a progress ring when the task is running.
+    /// </summary>
+    [ObservableProperty]
+    private bool _shouldShowProgressRing;
+
+    /// <summary>
+    /// The status symbol icon is the red, green, or yellow icon the is next to a task when it is completed.
```
Comment (krschau) [inline tools/SetupFlow/DevHome.SetupFlow/ViewModels/LoadingMessageViewModel.cs:27]: Nit
```suggestion
    /// The status symbol icon is the red, green, or yellow icon that is next to a task when it has been completed.

```

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/LoadingViewModel.cs [hunk 3]
```diff
@@ -50,6 +50,14 @@ public partial class LoadingViewModel : SetupPageViewModelBase
 
     private int _retryCount;
 
+    /// <summary>
+    /// A business rule for the loading screen is that all "executing" messages should always appear at the bottom
+    /// of the list of messages.  To enable this behavior all "finished" messages and "executing"-but-done messages
+    /// need to be inserted at the index before the first executing message.
+    /// Keep track this is used to count backwards from Messages.count to get the index to add messages to.
```
Comment (krschau) [inline tools/SetupFlow/DevHome.SetupFlow/ViewModels/LoadingViewModel.cs:57]: Nit: I think there are some words missing around "Keep track this"

File: tools/SetupFlow/DevHome.SetupFlow/Views/LoadingView.xaml [hunk 4]
```diff
@@ -20,7 +20,6 @@
                 <ResourceDictionary Source="ms-appx:///DevHome.SetupFlow/Styles/SetupFlowStyles.xaml" />
                 <ResourceDictionary>
                     <converters:BoolToVisibilityConverter x:Key="NegatedBoolToVisibilityConverter" TrueValue="Collapsed" FalseValue="Visible"/>
```
Comment (krschau) [inline tools/SetupFlow/DevHome.SetupFlow/Views/LoadingView.xaml:22]: I believe you can remove this one too.

**Comment by chenyingMS** [comment]
I might miss somewhere since I didn't find the update of RepoTool_AddRepos_Event from measure to critical (https://github.com/microsoft/devhome/issues/1308). Please help to update this. Thanks a lot!

File: telemetry/DevHome.Telemetry/Telemetry.cs (hunk 1 above)
Comment (dhoehna) [reply telemetry/DevHome.Telemetry/Telemetry.cs:184]: removed.

**Comment by dhoehna** [comment]
> RepoTool_AddRepos_Event

[The event is marked critical](https://github.com/microsoft/devhome/blob/user/dhoehna/LoggingStyleMessages/tools/SetupFlow/DevHome.SetupFlow/Views/RepoConfigView.xaml.cs#L88)

**Comment by chenyingMS** [comment]
> > RepoTool_AddRepos_Event
> 
> [The event is marked critical](https://github.com/microsoft/devhome/blob/user/dhoehna/LoggingStyleMessages/tools/SetupFlow/DevHome.SetupFlow/Views/RepoConfigView.xaml.cs#L88)

Thanks!

**Review by krschau — APPROVED** [review]

**Review by EricJohnson327 — APPROVED** [review]

**Pull request merged**